private/bufpkg/bufstudioagent/bufstudioagent_test.go

```diff
@@ -141,6 +143,7 @@ func testPlainPostHandlerErrors(t *testing.T, upstreamServer *httptest.Server) {
 			nil,
 		),
 	)
+	agentServer.Client().Timeout = time.Second
```

Comment:
If we didn't do this, tests would hang waiting for the http client to wait for a response. Default is >30s.

This test is still flaky because sometimes we were getting:
```
waiting before doing the request
signal sent
unblocked, starting request
connection arrived
connection closed
... timeout
```

Which meant that the http client was the one hanging out. After adding this shorter timeout, is still flaky, but failures are:

```
waiting before doing the request
signal sent
unblocked, starting request
connection arrived
connection closed
request completed
--- FAIL: TestPlainPostHandlerH2C (1.05s)
    --- FAIL: TestPlainPostHandlerH2C/invalid_upstream (1.00s)
        /Users/jfigueroa/dev/buf/buf/private/bufpkg/bufstudioagent/bufstudioagent_test.go:221: 
            	Error Trace:	/Users/jfigueroa/dev/buf/buf/private/bufpkg/bufstudioagent/bufstudioagent_test.go:221
            	Error:      	Received unexpected error:
            	            	Post "https://127.0.0.1:62351": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
            	Test:       	TestPlainPostHandlerH2C/invalid_upstream
FAIL
```

We suspect the failure comes from [this defer](https://github.com/bufbuild/connect-go/blob/713199600fa7bc3a47957c910ddde9a85253036f/duplex_http_call.go#L240) never being triggered. cc @akshayjshah @pkwarren